rust/agents/relayer/src/msg/gas_payment/mod.rs

```diff
@@ -29,11 +29,19 @@ pub trait GasPaymentPolicy: Debug + Send + Sync {
 #[derive(Debug)]
 pub struct GasPaymentEnforcer {
```

Comment:
I think as part of making this work with the pending changes in #1658 it will make the most sense to have multiple policies defined by whitelists and as to which to use. My thought is instead of doing whitelist/blacklist have a list of policies in order, and whitelist and then go in order until it matches the whitelist. The last should be a default that accepts everything.

Part of why this is how I think we will want to do it is instead of returning a bool it will start returning an allowed gas amount, so it makes sense to allow a policy to define that, even if it is just returning the estimated amount.

These changes do not need to happen in this PR but I think it is worth thinking about how that might change the configuration.

Reply:
To err on the side of making as few changes as needed, are you ok with merging as is?
I don't feel super strongly about how we do it, but having a general whitelist (or having it mean "use the gas enforcement policy None if it's a match") feels the simplest to reason about and like a small surface area of changes

Reply:
As-is is good by me, partially just spit-balling about future options.